### ISSUE TITLE
Problem with arduino controllers

### ISSUE THREAD

**Issue by madebytoilets** [TrueOpenVR/TrueOpenVR-DIY#15]
So i made arduino controllers using arduino uno, connected all buttons, mpu 6050 flashed the code. I changed in true open vr sethings to arduino controllers. And nothing happens. I dont have any idea. Please help me.

**Comment by r57zone** [comment]
Do you have any helmet connected?
Have you changed the com ports in the config file?
If the helmet is not connected, you need to make a splitter plugin, select emulation for the helmet and arduinocontrollers for the controllers in it.
You can also check the work in the GetData program.

**Comment by madebytoilets** [comment]
What is splitter plugin and helmet?
I tried with GetData program but again, controllers are not working

**Comment by r57zone** [comment]
![](https://user-images.githubusercontent.com/9499881/209814906-6879b9e5-51f1-401c-a0f1-b54e41df96c2.png)
![](https://user-images.githubusercontent.com/9499881/209814940-b2ddbede-7d50-4f22-8cd8-8e82357b1930.png)
![](https://user-images.githubusercontent.com/9499881/209814989-4a9d2640-793d-4830-8316-efc6e289bc9a.png)

**Comment by madebytoilets** [comment]
Still not working 
![image](https://user-images.githubusercontent.com/121180265/209815991-24d59ba6-2de7-4a36-9055-ea7a05a2e3d2.png)
and steam vr doesnt detect them 
![image](https://user-images.githubusercontent.com/121180265/209816150-d8d74cd5-edfa-40ef-ba68-3495634968ce.png)

**Comment by r57zone** [comment]
Have you correctly specified the monitor resolution, com ports?
Have you configured your SteamVR config?

**Comment by madebytoilets** [comment]
yeah, i tried with keyboard driver and it works perfectly. And i configured steam vr. Monitor solution is great, and com ports too

**Comment by r57zone** [comment]
it means that the wrong Arduino sketch may be flashed, I don’t know, but it looks like the driver does not see the com ports, otherwise there would be controllers icons

**Comment by madebytoilets** [comment]
wait, i tried now, and he showed me up the controllers on steam vr, but they are not working

**Comment by madebytoilets** [comment]
i dont know why, but when i connect arduino controllers, they looks like connected, but any respond, even if i press button on them

**Comment by wohdek21** [comment]
I got error like this, is it solved?

**Comment by wohdek21** [comment]
> i dont know why, but when i connect arduino controllers, they looks like connected, but any respond, even if i press button on them

i got error like this my controller working by bluetooth and showing that him online but no respond

**Comment by madebytoilets** [comment]
> I got error like this, is it solved?

nope

**Comment by madebytoilets** [comment]
> > i dont know why, but when i connect arduino controllers, they looks like connected, but any respond, even if i press button on them
> 
> i got error like this my controller working by bluetooth and showing that him online but no respond

I just gave up

**Comment by wohdek21** [comment]
bruh

**Comment by wohdek21** [comment]
if i will find a solution i will write here for everyone

**Comment by wohdek21** [comment]
i wrote email to owner and he is not responding now

**Comment by madebytoilets** [comment]
give him time

**Comment by madebytoilets** [comment]
> i wrote email to owner and he is not responding now

I tried to debug it but the exe is compiled. @r57zone if u can post decompiled code, I would look into it (:

**Comment by wohdek21** [comment]
> give him time

Ok

**Comment by wohdek21** [comment]
Wait I got that error with connected controllers but no respond but what if we just connect them to PSMoveService?

**Comment by wohdek21** [comment]
I will try this later

**Comment by r57zone** [comment]
If the controllers appear, then everything is working. It remains to flash the correct Arduino [sketch with a compatible output](https://github.com/TrueOpenVR/TrueOpenVR-DIY/tree/master/Controllers/Arduino) (Serial.write ...).
All sources are published, GetData sources is in TrueOpenVR-Samples

**Comment by wohdek21** [comment]
ok i tried to change SerialBT to Serial and it's not working now

**Comment by wohdek21** [comment]
Still not working but I flashed controller.ino file because sketch with mpu6050 is not flashing at the moment with error

**Comment by wohdek21** [comment]
OK what do I need to do for repairing controllers? @r57zone

**Comment by wohdek21** [comment]
nvm it's working now

**Comment by wohdek21** [comment]
no it's not
